deepfence_server/controls/agent.go

```diff
@@ -353,9 +353,10 @@ func ExtractStoppingAgentScans(ctx context.Context, nodeId string,
 	defer tx.Close()
 
 	r, err := tx.Run(`MATCH (s) -[:SCHEDULED]-> (n:Node{node_id:$id})
-		WHERE s.status = '`+utils.SCAN_STATUS_CANCEL_PENDING+`'
+		WHERE s.stop_requested=true
 		WITH s LIMIT $max_work
-        SET s.status = '`+utils.SCAN_STATUS_CANCELLING+`', s.updated_at = TIMESTAMP()
+        SET s.status = '`+utils.SCAN_STATUS_CANCELLING+`', 
+		s.updated_at = TIMESTAMP(), s.stop_requested=false
```

Comment:
Why are we introducing this flag?
Previous behavior is more aligned with the whole status workflow:
- Starting a scan: Starting, In progress
- Stopping a scan: Cancelling, Cancelled

No need to fork and have a `stop_requested` flag now

Reply:
Stop functionality by relying on the scan status will not work completely. 
Reason being, an already running scan will continuously emit "IN PROGRESS" asynchronously
Hence there's a chance we might overwrite the "CANCEL_PENDING" status with IN PROGRESS" even before we check this scan.
This will result in the stop never getting triggered.
Hence we would need this `stop_requested` flag for deterministic behavior.

Reply:
That's a good point, but the flag is not the solution: we need to set some priority when setting the status
The same way that if IN PROGRESS come after a COMPLETE we should not "uncomplete" the scan.
Let's make it consistent across the whole scan system

Reply:
We have some guarantees for the statuses:
* IN PROGRESS cannot come in before the "STARTING". 
* IN PROGRESS cannot come after the "COMPLETED" , "ERROR" or  "CANCELLED" state (terminal state). 
If these 2 are violated, we have a bug in the code and need to be fixed :) 

Also, the IN_PROGRESS is used at the kafka ingester for updating the db. Making any change there is not ideal as it's going to complicate the logic. Also, that's the hot path and might have performance impact.

Reply:
Reverted the flag change as we will be adding the logic to handle this in the status ingestion.